src/family.toit

```diff
@@ -0,0 +1,412 @@
+Real-time clock (RTC).
+
+Also includes DS2415.
+*/
+FAMILY_DS1904 ::= 0x24
+
+/** Smart battery monitor. */
+FAMILY_DS2438 ::= 0x26
+
+/** RTC with interrupt. */
+FAMILY_DS2417 ::= 0x27
+
+/** Programmable resolution thermometer. */
+FAMILY_DS18B20 ::= 0x28
+
+/** 4-channel addressable switch. */
+FAMILY_DS2408 ::= 0x29
+
+/** 1-channel digital potentiometer. */
+FAMILY_DS2890 ::= 0x2C
+
+/**
+1024-bit EEPROM memory.
+
+Also includes DS2431.
+*/
+FAMILY_DS1972 ::= 0x2D
+
+/** Battery management. */
+FAMILY_DS2770 ::= 0x2E
+
+/** High-precision Li+ battery monitor. */
+FAMILY_DS2760 ::= 0x30
+
+/** Single-cell rechargeable lithium protection. */
+FAMILY_DS2720 ::= 0x31
+
+/** Battery fuel gauge. */
+FAMILY_DS2780 ::= 0x32
+
+/**
+1128-bit iButton with SHA-1 engine.
+
+Also includes DS2432.
+*/
+FAMILY_DS1961S ::= 0x33
+
+/** Battery pack authentication. */
+FAMILY_DS2703 ::= 0x34
+
+/** High-accuracy battery fuel gauge. */
+FAMILY_DS2755 ::= 0x35
+
+/** High-precision coulomb counter. */
+FAMILY_DS2740 ::= 0x36
+
+/** Password-protected 32KB (bytes) EEPROM. */
+FAMILY_DS1977 ::= 0x37
+
+/** 2-channel addressable switch. */
+FAMILY_DS2413 ::= 0x3A
+
+/**
+Programmable resolution thermometer.
+
+Also includes MAX31826 (temperature sensor with 1Kb lockable EEPROM).
+*/
+FAMILY_DS1825 ::= 0x3B
+
+/** 1-cell or 2-cell battery fuel gauge. */
+FAMILY_DS2781 ::= 0x3D
+
+/**
+High-capacity Thermochron (temperature) and Hygrochron™ (humidity) loggers.
+
+Also includes DS1922T, DS1923, and DS2422
+*/
+FAMILY_DS1922L ::= 0x41
+
+/** Programmable resolution digital thermometer with sequenced detection and PIO. */
+FAMILY_DS28EA00 ::= 0x42
+
+/** 20Kb 1-Wire EEPROM. */
+FAMILY_DS28EC20 ::= 0x43
+
+/** SHA-1 authenticator. */
+FAMILY_DS28E10 ::= 0x44
+
+/** Multichemistry battery fuel gauge. */
+FAMILY_DS2751 ::= 0x51
+
+/** Environmental sensor. */
+FAMILY_EDS00XX ::= 0x7E
+
+/** USB ID. */
+FAMILY_DS2490 ::= 0x81
+
+/** Multi iButton with three 384-bit fields of password-protected RAM. */
+FAMILY_DS1425 ::= 0x82
+
+/** UniqueWare™ iButton with 1024 bits EPROM. */
+FAMILY_DS1982U ::= 0x89
+
+/** UniqueWare™ iButton with 16Kb EPROM. */
+FAMILY_DS1985U ::= 0x8B
+
+/** UniqueWare™ iButton with 64Kb EPROM. */
+FAMILY_DS1986U ::= 0x8F
+
+/** Shaft rotation sensor. */
+FAMILY_MRS001 ::= 0xA0
+
+/** Vibration sensor. */
+FAMILY_MVM001 ::= 0xA1
+
+/** AC current sensor. */
+FAMILY_MCMC001 ::= 0xA2
+
+/** Infrared temperature sensor. */
+FAMILY_MTS017 ::= 0xA6
+
+/** Thermocouple converter. */
+FAMILY_MTC001 ::= 0xB1
+
+/** Analog input module. */
+FAMILY_MAM001 ::= 0xB2
+
+/** Thermocouple converter. */
+FAMILY_MTC002 ::= 0xB3
+
+/** UV index sensor. */
+FAMILY_UVI_01 ::= 0xEE
+
+/** Moisture hub. */
+FAMILY_MOISTURE_HUB ::= 0xEF
+
+/**
+Programmable microprocessor.
+
+Also includes BAE0911.
+*/
+FAMILY_BAE0910 ::= 0xFC
+
+/** LCD. */
+FAMILY_LCD ::= 0xFF
+
+
+family_id --device_id/int -> int:
+    return device_id & 0xFF
+
+family_to_string family_id/int -> string:
```

Comment:
Should this be an array of strings?

Reply:
Now that it's already a sequence of ifs, might as well leave it.
Eventually this should be a 'switch'.